README.md

```diff
@@ -137,6 +137,26 @@ You can also use the ND HTTPAPI connection plugin with your cisco.mso Ansible co
       state: query
 ```
 
+User API Key authorization is also supported in the ND HTTPAPI connection plugin. Use the `ansible_httpapi_session_key` option to specify the key instead of a password. The `ansible_httpapi_session_key` option takes precedence over the `ansible_password` option if defined. If authorization fails using the API Key, the plugin will fallback to using the password. The session key option must be defined as a dictionary. The dictionary can either be formated in two ways:
+
+1. Using a single key-value dictionary eg. `{"keyname": "<APIKEY>"}`. Only the dictionary value is used with the `ansible_user` option to format the ND authorization header.
+
+2. Providing the ND authorization header manually eg. `{"X-Nd-Username": "admin", "X-Nd-Apikey": "<APIKEY>"}`. The `ansible_httpapi_session_key` option will be used as-is and the `ansible_user` option is ignored.
+
+See the [Authorization Using API Key](https://developer.cisco.com/docs/nexus-dashboard/latest/getting-started/#authorization-using-api-key) documentation for more information.
```

Comment:
I think it would really benefit the user by giving them the option of providing the session_key in only one format without confusing them too much.

Can we change the session key such that the key is the Username and the value is the API key?
```
ansible_httpapi_session_key={"<USERNAME>": "<APIKEY>"}
```

Reply:
I don't believe two formats is too many for our users and the first format is preferred over the second. I added the second option as a backup format and this is why I initially kept it undocumented. The second is a little more complicated to enter and will ignore any `ansible_user` options.

I chose to use the first format so it doesn't require a username in the key while the `ansible_user` option is retained. Additionally if the `username` option is used or changed within a task it will be reflected in the key authorization header. If we use the username in the `session_key` all other username options would be ignored.

Reply:
I still think we don't need "X-Nd-Username" and "X-Nd-Apikey" because they are already hardcoded in your PR and providing one option makes more sense -> ansible_httpapi_session_key={"<USERNAME>": "<APIKEY>"}
I'll approve the PR though.

Reply:
I think we can keep user being provided with the ansible_user variable in case someone does not use the full header in ansible_httpapi_session_key.